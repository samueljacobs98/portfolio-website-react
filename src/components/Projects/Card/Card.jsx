import "./Card.scss";
import { useState } from "react";

const Card = ({ data, handleClick, index, projectToShow }) => {
  const [windowIsBig, setWindowIsBig] = useState(window.innerWidth > 1060);
  const handleResize = () => {
    setWindowIsBig(window.innerWidth > 1060);
  };

  window.addEventListener("resize", handleResize);

  const { title, shortDesc, skills, id } = data;

  const isActive = id === projectToShow;
  const cardClassName = isActive ? "card card--active" : "card";

  const href = windowIsBig ? "#projects" : "#project";

  return (
    <a href={href} className={cardClassName} onClick={() => handleClick(index)}>
      <p className="card__title">{title}</p>
      <p className="card__brief">{shortDesc}</p>
      <div className="card__line"></div>
      <div className="card__skills">
        <div className="card__skills-left">
          <p className="skill skill1">{skills[0]}</p>
          <p className="skill skill2">{skills[1]}</p>
        </div>
        <div className="card__skills-right">
          <p className="skill skill3">{skills[2]}</p>
          <p className="skill skill4">{skills[3]}</p>
        </div>
      </div>
    </a>
  );
};

export default Card;
