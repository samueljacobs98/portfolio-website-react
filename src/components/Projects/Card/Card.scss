@use "../../../assets/scss/variables" as *;

.card {
  background-color: $colour-background;
  width: $card-width;
  min-height: $card-min-height;
  padding: $padding-card;
  text-align: center;
  border: 2px solid $color-white;
  &__title {
    font-size: $font-size;
    text-decoration: underline;
  }
  &__line {
    border-bottom: 2px solid $color-white;
  }
  &__title,
  &__brief,
  &__line {
    margin-bottom: $margin-card-text;
  }
  &__skills {
    display: flex;
    justify-content: space-between;
    &-left {
      text-align: left;
    }
    &-right {
      text-align: right;
    }
  }
  outline: 0px solid $color-white;
  transition: outline 0.75s ease-in-out;
}

.card:hover {
  cursor: pointer;
  outline: 4px solid $color-white;
}
