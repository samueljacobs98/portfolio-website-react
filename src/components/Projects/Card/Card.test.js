import { render } from "@testing-library/react";
import Card from "./Card";

test("should match snapshot for active card", () => {
  const data = { title: "", shortDesc: "", skills: ["", ""], id: 1 };
  const { container } = render(
    <Card data={data} handleClick={null} index={null} projectToShow={1} />
  );
  expect(container).toMatchSnapshot();
});

test("should match snapshot for inactive card", () => {
  const data = { title: "", shortDesc: "", skills: ["", ""], id: 1 };
  const { container } = render(
    <Card data={data} handleClick={null} index={null} projectToShow={2} />
  );
  expect(container).toMatchSnapshot();
});
