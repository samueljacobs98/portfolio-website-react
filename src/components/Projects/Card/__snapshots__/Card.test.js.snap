// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for active card 1`] = `
<div>
  <div
    class="card card--active"
  >
    <p
      class="card__title"
    />
    <p
      class="card__brief"
    />
    <div
      class="card__line"
    />
    <div
      class="card__skills"
    >
      <div
        class="card__skills-left"
      >
        <p
          class="skill skill1"
        />
        <p
          class="skill skill2"
        />
      </div>
      <div
        class="card__skills-right"
      >
        <p
          class="skill skill3"
        />
        <p
          class="skill skill4"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`should match snapshot for inactive card 1`] = `
<div>
  <div
    class="card"
  >
    <p
      class="card__title"
    />
    <p
      class="card__brief"
    />
    <div
      class="card__line"
    />
    <div
      class="card__skills"
    >
      <div
        class="card__skills-left"
      >
        <p
          class="skill skill1"
        />
        <p
          class="skill skill2"
        />
      </div>
      <div
        class="card__skills-right"
      >
        <p
          class="skill skill3"
        />
        <p
          class="skill skill4"
        />
      </div>
    </div>
  </div>
</div>
`;
